### DISCUSSION TITLE
Fullstack next.js

### DISCUSSION THREAD

**Discussion by rajesh99** [Southclaws/storyden#179] [General]
I do nodejs and java and GCP stuff.  Hence, we don't want to enter into golang stack.

I want to replace golang stack with nextjs backend.

Ideas on how to do?

**Comment by Southclaws** [comment]
Storyden can be deployed as a static binary (zero external dependencies) or via a Docker image so if you’re not planning to make source changes it shouldn’t matter the language it’s written in (and if you are seeking a deployment I am happy to support your usage via email/whatsapp/discord)

Since your recent question about GCP I am working on documentation for a GCP deployment and list GCP as an officially supported platform.

Replacing the backend would be pretty much a complete rewrite as the majority of the functionality is held in the backend itself, the front end is just a simple Next.js application.

You could do it in theory and use the OpenAPI spec to stub endpoints in JS but you’d still be reimplementing the entire (or a subset) Go codebase which would be quite a time cost.

I’m interested to hear more about your use case and what about the current architecture is making you hesitant to deploy.

**Comment by rajesh99** [comment]
I am looking to offer as part of saas.  So, we have to have multi-tenancy and user management.  
I am also looking at providing good auth solutions and that can be integrated into Google Workspace marketplace.  I have plans to add more knowledge-sharing apps.  Hence, I am looking for nodejs solutions and dev env, as this is our preferred choice among my small developers.  
While looking out, Storyden got my attention and I was exploring.
Thanks for your support.

**Reply by Southclaws** [reply]
Interesting! Well I wish you the best on your endeavour, if you use Storyden then support and updates are always here. And if you choose to modify the code for your SaaS as long as you're abiding by the MPL-2.0 license by publishing the modifications, it's all good!

I'm interested to follow your journey so feel free to share updates in [our community](https://discord.gg/XF6ZBGF9XF)
